### PR TITLE
DSD-854: Updating all console warnings with NYPL branding

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-854](https://jira.nypl.org/browse/DSD-854)

## This PR does the following:

- Updates all console warnings with NYPL branding.
- Some of the language was updated so we don't have "we" or "you" in the message and it can be more direct and active.
- Adds missing tests for some console warning messages.

## How has this been tested?

Unit tests

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr917-idbgjglu5gydpdezojbhbabq0fuur1n3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623dd5c1628e351b6b786e54

File: src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -327,12 +327,12 @@ const DatePicker = React.forwardRef<TextInputRefType, DatePickerProps>(
...
     if ((ref && !nameFrom) || (refTo && !nameTo)) {
       console.warn(
-        "A `ref` or `refTo` prop was passed but not the equivalent `nameFrom` or `nameTo` prop."
+        "NYPL Reservoir DatePicker: a `ref` or `refTo` prop was passed but not the equivalent `nameFrom` or `nameTo` prop."
       );
     }
     if (onChange && (ref || refTo || nameFrom || nameTo)) {
       console.warn(
-        "React `ref` props were passed and an `onChange` prop as well. Use whichever is best for your app but not both."
+        "NYPL Reservoir DatePicker: React `ref` props were passed and an `onChange` prop as well. Use whichever is best for your app but not both."
```
Comment (bigfishdesign13) [inline src/components/DatePicker/DatePicker.tsx]: Could be clearer.
```
React `ref` and `onChange` props were both passed. Use…
```

Or should should `ref` be referred to in a singular sense?  `prop` vs `props`?

File: src/components/Hero/Hero.tsx [hunk 2]
```diff
@@ -68,35 +68,37 @@ export default function Hero(props: React.PropsWithChildren<HeroProps>) {
   if (heroType === HeroTypes.Primary) {
     if (!backgroundImageSrc) {
       console.warn(
-        `Warning: it is recommended to use the "backgroundImageSrc" prop for PRIMARY hero.`
+        `NYPL Reservoir Hero: it is recommended to use the "backgroundImageSrc" prop for PRIMARY hero.`
```
Comment (bigfishdesign13) [inline src/components/Hero/Hero.tsx]: `PRIMARY` should not be capitalized, but I am not exactly sure what it should be replaced with.  `HeroTypes.Primary`?  And this change is applicable to all references to the `Hero` types.

File: src/components/Hero/Hero.tsx [hunk 3]
```diff
@@ -68,35 +68,37 @@ export default function Hero(props: React.PropsWithChildren<HeroProps>) {
...
     if (image) {
       console.warn(
-        `Warning: the "image" prop has been passed, but PRIMARY hero will not use it.`
+        `NYPL Reservoir Hero: the "image" prop has been passed, but PRIMARY hero will not use it.`
       );
     }
   } else if (locationDetails) {
     console.warn(
-      `Warning: Please provide "locationDetails" only to PRIMARY hero.`
+      `NYPL Reservoir Hero: Please provide "locationDetails" only to PRIMARY hero.`
```
Comment (bigfishdesign13) [inline src/components/Hero/Hero.tsx]: Could be clearer.
```
NYPL Reservoir Hero: the "locationDetails” prop should only be used with HeroTypes.Primary.
```

File: src/components/HorizontalRule/HorizontalRule.tsx [hunk 4]
```diff
@@ -27,7 +27,7 @@ export default function HorizontalRule(props: HorizontalRuleProps) {
 
   if (height.endsWith("%")) {
     console.warn(
-      "`HorizontalRule`: For the `height` prop, use a whole number, a `px`" +
+      "NYPL Reservoir HorizontalRule: For the `height` prop, use a whole number, a `px`" +
```
Comment (bigfishdesign13) [inline src/components/HorizontalRule/HorizontalRule.tsx]: Could be clearer.
```
NYPL Reservoir HorizontalRule: An invalid value was passed for the `height` prop, so the default value of 2px will be used.  A valid value should be a whole number, a `px` value, a `em` value, or a `rem` value.
```

File: src/components/Icons/Icon.tsx [hunk 5]
```diff
@@ -77,12 +77,12 @@ export default function Icon(props: React.PropsWithChildren<IconProps>) {
   // Component prop validation
   if (name && children) {
     console.warn(
-      "Icon accepts either a `name` prop or an `svg` element child, not both."
+      "NYPL Reservoir Icon: pass in either a `name` prop or an `svg` element child, not both."
```
Comment (bigfishdesign13) [inline src/components/Icons/Icon.tsx]:
```
NYPL Reservoir Icon: Pass in either a `name` prop or an `svg` element child. Do not pass both.
```

File: src/components/List/List.tsx [hunk 6]
```diff
@@ -60,13 +60,13 @@ export default function List(props: React.PropsWithChildren<ListProps>) {
   // prop must be used.
   if (children && (listItems || listItems?.length > 0)) {
     console.warn(
-      "Pass in either `<li>`, `<dt>`, or `<dd>` children or use the `listItems` data prop, but don't use both."
+      "NYPL Reservoir List: pass in either `<li>`, `<dt>`, or `<dd>` children or use the `listItems` data prop, but don't use both."
```
Comment (bigfishdesign13) [inline src/components/List/List.tsx]:
```
NYPL Reservoir List: Pass in either `<li>`, `<dt>`, or `<dd>` children or use the `listItems` data prop. Do not use both.
```

File: src/components/Logo/Logo.tsx [hunk 7]
```diff
@@ -55,12 +55,12 @@ export default function Logo(props: React.PropsWithChildren<LogoProps>) {
   // Component prop validation
   if (name && children) {
     console.warn(
-      "Logo accepts either a `name` prop or an `svg` element child. It can not accept both."
+      "NYPL Reservoir Logo: accepts either a `name` prop or an `svg` element child. It can not accept both."
```
Comment (bigfishdesign13) [inline src/components/Logo/Logo.tsx]:
```
NYPL Reservoir Logo: Pass either a `name` prop or an `svg` element child. Do not pass both.
```

File: src/components/ProgressIndicator/ProgressIndicator.tsx [hunk 8]
```diff
@@ -60,7 +60,7 @@ const ProgressIndicator: React.FC<ProgressIndicatorProps> = (
   let finalValue = value;
   if (finalValue < 0 || finalValue > 100) {
     console.warn(
-      "ProgressIndicator: pass in a `value` between 0 and 100. Defaulting to 0."
+      "NYPL Reservoir ProgressIndicator: pass in a `value` between 0 and 100. Defaulting to 0."
```
Comment (bigfishdesign13) [inline src/components/ProgressIndicator/ProgressIndicator.tsx]:
```
NYPL Reservoir ProgressIndicator: An invalid value was passed for the `value` prop, so 0 will be used. A valid value should be a number between 0 and 100.
```

File: src/components/StructuredContent/StructuredContent.tsx [hunk 9]
```diff
@@ -113,7 +113,7 @@ export default function StructuredContent(
 
   if (hasImage && !imageAlt) {
     console.warn(
-      "StructuredContent: `imageAlt` prop is required when using an image."
+      "NYPL Reservoir StructuredContent: `imageAlt` prop is required when using an image."
```
Comment (bigfishdesign13) [inline src/components/StructuredContent/StructuredContent.tsx]:
```
NYPL Reservoir StructuredContent: The `imageAlt` prop is required when using an image.
```

File: src/components/VideoPlayer/VideoPlayer.tsx [hunk 10]
```diff
@@ -83,22 +83,22 @@ export default function VideoPlayer(
...
   if (!embedCodeFinal && !videoType && !videoId) {
     console.warn(
-      "VideoPlayer requires either the `embedCode` prop or both the `videoType` and `videoId` props."
+      "NYPL Reservoir VideoPlayer: requires either the `embedCode` prop or both the `videoType` and `videoId` props."
     );
     isInvalid = true;
   } else if (!embedCodeFinal && !videoType) {
     console.warn(
-      "VideoPlayer also requires the `videoType` prop. You have only set the `videoId` prop."
+      "NYPL Reservoir VideoPlayer: requires the `videoType` prop. Only the `videoId` prop was set."
```
Comment (bigfishdesign13) [inline src/components/VideoPlayer/VideoPlayer.tsx]:
```
NYPL Reservoir VideoPlayer: The `videoType` prop is also required. Only the `videoId` prop was set.
```

File: src/components/VideoPlayer/VideoPlayer.tsx [hunk 11]
```diff
@@ -83,22 +83,22 @@ export default function VideoPlayer(
...
     );
     isInvalid = true;
   } else if (!embedCodeFinal && !videoId) {
     console.warn(
-      "VideoPlayer also requires the `videoId` prop. You have only set the `videoType` prop."
+      "NYPL Reservoir VideoPlayer: requires the `videoId` prop. Only the `videoType` prop was set."
```
Comment (bigfishdesign13) [inline src/components/VideoPlayer/VideoPlayer.tsx]:
```
NYPL Reservoir VideoPlayer: The `videoId` prop is also required. Only the `videoType` prop was set.
```

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
The writing style changes throughout the warning messages.  I tried to address those where the differences were significant, but overall there are still a number that I did not address.  I think we should go the route of full sentences, or somethign very close.  What do you think?

**Comment by EdwinGuzman** [comment]
Sure, full sentences are fine. I was being a bit lazy and leaving the text as is. I'll update the rest.

**Comment by EdwinGuzman** [comment]
Updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr917-idbgjglu5gydpdezojbhbabq0fuur1n3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623dd5c1628e351b6b786e54

File: src/components/Heading/Heading.tsx [hunk 12]
```diff
@@ -64,17 +64,15 @@ function Heading(props: React.PropsWithChildren<HeadingProps>) {
...
 
   if (React.Children.count(props.children) > 1) {
-    const children = React.Children.map(
-      props.children,
-      (child) => (child as JSX.Element).type
-    );
     // Catching the error because React's error isn't as helpful.
     throw new Error(
-      `Please only pass one child into Heading, got ${children.join(", ")}`
+      "NYPL Reservoir Heading: Only pass one child into Heading; got multiple children."
```
Comment (bigfishdesign13) [inline src/components/Heading/Heading.tsx]:
```
NYPL Reservoir Heading: Only pass one child into Heading. Multiple children have been passed.
```
Or even just this…
```
NYPL Reservoir Heading: Only pass one child into Heading.
```

File: src/components/Pagination/Pagination.tsx [hunk 13]
```diff
@@ -64,13 +64,15 @@ const Pagination: React.FC<PaginationProps> = (props: PaginationProps) => {
   }
   if (getPageHref && onPageChange) {
     console.warn(
-      "NYPL Reservoir Pagination: Props for both `getPageHref` and `onPageChange` are passed. Will default to using `getPageHref`."
+      "NYPL Reservoir Pagination: Props for both `getPageHref` and " +
+        "`onPageChange` are passed. Will default to using `getPageHref`."
```
Comment (bigfishdesign13) [inline src/components/Pagination/Pagination.tsx]:
```
… The component will default to using `getPageGref`.
```

File: src/components/Select/Select.tsx [hunk 14]
```diff
@@ -108,13 +108,17 @@ const Select = React.forwardRef<
 
   if (React.Children.count(children) > 10) {
     console.warn(
-      "NYPL DS recommends that <select> fields have no more than 10 options; an auto-complete text input is a good alternative for 11 or more options."
+      "NYPL Reservoir Select: It is recommended that <select> fields have " +
```
Comment (bigfishdesign13) [inline src/components/Select/Select.tsx]: After hearing about Bill’s woes with these warning messages, let’s remove both of these and instead include something about best practices on the `Select` story page.

File: src/components/Tabs/Tabs.tsx [hunk 15]
```diff
@@ -223,7 +225,8 @@ function Tabs(props: React.PropsWithChildren<TabsProps>) {
 
   if (children && contentData?.length) {
     console.warn(
-      "Tabs: Only pass children or data in the `data` props but not both."
+      "NYPL Reservoir Tabs: Only pass children or data in the `contentData` " +
```
Comment (bigfishdesign13) [inline src/components/Tabs/Tabs.tsx:228]:
```
NYPL Reservoir Tabs: Only pass children or data in the `contentData` prop. Do not pass both.
```

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Looking very good.  Just a couple of small adjustments for clarity.

**Review by bigfishdesign13 — APPROVED** [review]
Nailed it!

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr917-idbgjglu5gydpdezojbhbabq0fuur1n3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623dd5c1628e351b6b786e54

**Pull request merged**